pool/src/price_provider.rs

```diff
@@ -0,0 +1,20 @@
+use price_feed_interface::PriceFeedClient;
+use soroban_sdk::{Address, Env};
+
+#[allow(dead_code)]
+pub struct PriceProvider<'a> {
+    feed: PriceFeedClient<'a>,
+}
+
+#[allow(dead_code)]
+impl PriceProvider<'_> {
+    pub fn new(env: &Env, feed_address: Address) -> Self {
+        let feed = price_feed_interface::PriceFeedClient::new(env, &feed_address);
+        Self { feed }
+    }
+
+    pub fn get_price(&self, asset: Address) -> Option<i128> {
```

Comment:
It could be better to provide price and decimals. You can wrap it in structure

Reply:
Extended with `AssetPrice`.